### ISSUE TITLE
Stone quarry stock issue

### ISSUE THREAD

**Issue by Truetobi91** [Sh0wdown/UnofficialCrusaderPatch#305]
I think that's a pretty common issue but I'll explain it for anyone who isn't aware of it.

Each stone quarry comes with a small 2x2 tiles big stock where the mined stoned is stocked. There are two problems with this stock: it can be destroyed by the AI when building something else on top of it (e.g. a wall) or it can be isolated when moat is being dugaround it. This renders the whole quarry useless and can heavily handicap an AI especially when they have few quarries to begin with or at the start of the game. 

Solution: AI should destroy the whole quarry and replace it if the quarry stock is not available or can't be reached independant on the options to disable the destruction of economy buildings under certain conditions or buildings that are unaccessable.

**Comment by DarkThief-Darek** [comment]
There is actually a bug (feature?) in the game, when the 2x2 stock pile is not accessible by the workers: The ox tether worker who tries to reach the inaccessible stock pile will teleport the stone from the stock pile into his ox tether, remove the time the worker needs to carry the stone.

Works for AIs as well as the players. 

After play testing on custom detailed maps and AI stone quarry placements I came to the conclusion that the AI build more inaccessible stone stock piles as you might think. But often due to the ox tether placement the workers will choose a closer, more available stock pile than the inaccessible one, which would trigger the bug.

**Comment by PitchNeeded** [comment]
Another issue is the AI not rebuilding, or being slow to rebuild damaged stone quarries late game when they are not getting any wood supply.

**Comment by Truetobi91** [comment]
@DarkThief-Darek 
I haven't seen this bug ever, do you know how to reproduce this for the AI?

**Comment by DarkThief-Darek** [comment]
@Truetobi91 You can reproduce it easily as a player. Just wall off your 2x2 stone stockpile.

For AIs: When the stone fields on the map are surrounded by a lot of uneven terrain or inaccessible map details like big dark rock types etc. the AI has the chance to build quarries where the 2x2 stockpile is located in an inaccessible spot (or walling it off with another quarry/castle walls etc.). If an ox tether is then placed near that quarry, you will discover the teleporting stones. The ox tether worker will always prioritize the quarry next to them (when they have stone in the stockpile)

**Comment by Truetobi91** [comment]
@DarkThief-Darek 

I was finally able to reproduce it and found out that there needs to be at least some space between the stock and obstructing blocks (walls; river; elevation; facing stone - all worked out). When I placed the obstructing blocks right next to the stock without any room, the bug would not appear.

**Comment by Halbarad1** [comment]
How about just putting the onto/ into the quarry? So the ox tether only has to reach the quarry. Could look like this (second quarry):
 
![Steinbruch stronghold](https://user-images.githubusercontent.com/44904062/55753356-60d28580-5a4a-11e9-8429-967df1f5d7bb.png)

**Comment by DarkThief-Darek** [comment]
@Halbarad1 Haha, this would have solved so many pathfinding and placement issues. Funny but effective idea! :D

**Comment by Truetobi91** [comment]
@Halbarad1 

Yes that could be a good solution for the AI. Whenever I play myself I like to place 1-2 Ox tethers right next to the stone stock for max efficiency which would be worse with the change BUT I'd take your solution any day if that fixes problems with the AI!

**Comment by Truetobi91** [comment]
@Sh0wdown 
How did you implement the Ox Tether Spam deactivation that basically sets the maximum number of Ox tethers to 10? Is it possible to edit this in the AIC or did you do this in other data files?

**Comment by Sh0wdown** [comment]
> @Sh0wdown
> How did you implement the Ox Tether Spam deactivation that basically sets the maximum number of Ox tethers to 10? Is it possible to edit this in the AIC or did you do this in other data files?

I did not limit them. I just stopped the AI from placing an ox tether every time they build a quarry.

**Comment by Truetobi91** [comment]
> 
> I did not limit them. I just stopped the AI from placing an ox tether every time they build a quarry.

Despite running different tests the AI always built exactly 10 ox tethers if there were enough stone quarries. The only somewhat fitting parameter in AIC is Unknown010 that ranges from 8 (Sultan) to 12 (Wazir) so I think that there is no AIC parameter that determines the maximum number of ox tethers. 

So there is either a global parameter that can be changed to increase the maximum number of ox tethers or we can start reducing AIs number of stone quarries since there isn't much efficiency in building more than 2, maybe 3 of them. The efficiency of ox tethers is somewhat decreased by positive fear factor but heavily cut by inefficient placement:
- unnecessary distance between quarry and tether
- tether placed on the other side of terrain
- quarries being build at different locations and ox tethers being distributed unevenly.
